pip/tests/test_project.py

```diff
@@ -51,11 +51,23 @@ def test_project_unreadable_source(qsharp) -> None:
     assert str(excinfo.value).find("OSError: could not read test.qs") != -1
 
 
+def test_project_dependencies(qsharp) -> None:
+    qsharp.init(project_root="/with_deps")
+    result = qsharp.eval("Test.CallsDependency()")
+    assert result == 4
+
+
+def test_project_circular_dependency_error(qsharp) -> None:
+    with pytest.raises(Exception) as excinfo:
+        qsharp.init(project_root="/circular")
+    assert str(excinfo.value).find("Circular dependency detected between") != -1
+
+
 memfs = {
```

Comment:
Would it make sense to have some fake github sources here to test that functionality in Python too?